### PR TITLE
ci: updated publish scripts to work the same way as the TERArium repo

### PR DESCRIPTION
**Pull request by dgauldie**
# Description

These CI scripts should work the same way as TERArium

Resolves #(issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] CI change

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide reproduction instructions and list any relevant details for your test configuration

- [ ] need to manually test 

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by YohannParis** [comment]
I don't understand the purpose of the bake file in this repo?
Shouldn't they go to the `.github` repo?

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -11,23 +11,83 @@ on:
...
-  # Call the Lint & Format workflow before publishing
-  verify:
+  # Run Global lint
+  lint:
     uses: ./.github/workflows/lint.yml
 
-  # Generate a more robust tag for the image
-  tag-generator:
-    name: Determine image tag
+  # Run yaml file lint
```
Comment (Nithos) [inline .github/workflows/publish.yaml:27]: Remove these ones, the main `lint` does them all here along with k8s. Terarium was split b/c it has a slightly different lint strategy (ie JS/TS linting etc..)

TLDR: remove `lint-yaml` `lint-docker`

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -11,23 +11,83 @@ on:
...
+    if: github.ref_name == 'main' || contains(github.event.ref, '/tags/')
+    permissions:
+      packages: write
+      contents: read
+
+    # this job depends on the following jobs before beginning
+    needs:
+      - targets
+      - lint
+      - lint-yaml
```
Comment (Nithos) [inline .github/workflows/publish.yaml:72]: remove yaml and docker once the above is done as well

File: .github/workflows/bake-publish.yml [hunk 3]
```diff
@@ -1,129 +0,0 @@
----
-# This is a reusable workflow for building and publishing Docker images
```
Comment (Nithos) [inline .github/workflows/bake-publish.yml:2]: This was meant as the reusable workflow... in theory Terarium should be using this but due to the extra builds that need to happen there I have not been able to do that yet.
IMO this is better if we just have docker files that need to be built... for Terarium ideally we would change to a multi build dockerfile that does the necessary builds within it rather than relying on external builds to happen and move the files over.

**Review by Nithos — CHANGES_REQUESTED** [review]
I think we should discuss this further.

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (dgauldie) [reply .github/workflows/publish.yaml]: fixed

**Comment by dgauldie** [comment]
this work is not needed
